File: src/projectile.py
#! /usr/bin/env python3

# projectile.py - analyze dragless projectile motion on Earth

from math import radians, sin

ANGLE = radians(float(input("Launch Angle    (deg) = ")))
SPEED =         float(input("Launch Speed    (m/s) = "))

# Standard acceleration due to gravity
GN = 9.80665

print(f"Flight Peak     (m)   = {pow(SPEED * sin(ANGLE), 2) / 2 / GN:g}")
print(f"Flight Range    (m)   = {pow(SPEED, 2) * sin(2 * ANGLE) / GN:g}")
print(f"Flight Duration (s)   = {2 * SPEED * sin(ANGLE) / GN:g}")

File: src/projectile.c
// projectile.c - analyze dragless projectile motion on Earth

#include <math.h>
#include <stdio.h>

// Full angle in radians
#define TAU 6.28318530717958647692

// Standard acceleration due to gravity
#define GN 9.80665

int main(void) {
  double angle, speed, yspeed;

  fputs("Launch Angle    (deg) = ", stdout);
  scanf("%lg", &angle);
  fputs("Launch Speed    (m/s) = ", stdout);
  scanf("%lg", &speed);

  // Convert the angle from degrees to radians
  angle *= TAU / 360;
  yspeed = speed * sin(angle);

  printf("Flight Peak     (m)   = %g\n", yspeed * yspeed / (2 * GN));
  printf("Flight Range    (m)   = %g\n", speed * speed * sin(2 * angle) / GN);
  printf("Flight Duration (s)   = %g\n", 2 * yspeed / GN);
}
